### PR TITLE
Add tests for the Command classes

### PR DESCRIPTION
**Pull request by hipio**
This is one of the two PR's containing tests for this library. The reason for breaking up these PR's is so that I can make sure Scrutinizer and Travis play nice (when it comes to code coverage).

### CODE REVIEW

File: src/Tests/AbstractCommandTestCase.php [hunk 1]
```diff
@@ -0,0 +1,307 @@
...
+
+    /** @var array<string, MockObject> */
+    private $mockServices;
+
+    /**
+     * Setup method for all the fixture. This method is called before a test is executed.
+     */
+    public function setUp()
+    {
+        parent::setUp();
```
Comment (Potherca) [inline src/Tests/AbstractCommandTestCase.php]: There is no parent setup.

File: src/Tests/AbstractCommandTestCase.php [hunk 2]
```diff
@@ -0,0 +1,307 @@
...
+
+        $this->buildMocks();
+
+        $this->setupApplication();
+        $this->setupDoctrine();
+        $this->setupServices();
+    }
+
+    /**
```
Comment (Potherca) [inline src/Tests/AbstractCommandTestCase.php]: Protected utility functions belong at the end of the file. Apart from getters/setters public api comes first.

File: src/Tests/Command/AbstractCommandTest.php [hunk 3]
```diff
@@ -0,0 +1,51 @@
...
+         * We are using this class to test the AbstractCommand class
+         * as we cannot use an abstract class directly
+         */
+        $this->command = new CreateApiTokenCommand();
+    }
+
+    /**
+     * @uses \DealerDirect\Security\API\Command\CreateApiTokenCommand
+     */
+    public function testCreateApiTokenCommandWithoutAnyInput()
```
Comment (Potherca) [inline src/Tests/Command/AbstractCommandTest.php]: Function name does not reflect expectation.

Suggestion: `testCreateApiTokenCommandShouldProtestWhenRunWithoutInput()`

File: src/Tests/Command/CreateApiTokenCommandTest.php [hunk 4]
```diff
@@ -0,0 +1,124 @@
...
+use Symfony\Component\Console\Input\StringInput;
+use Symfony\Component\Console\Output\StreamOutput;
+
+/**
+ * @coversDefaultClass \DealerDirect\Security\API\Command\CreateApiTokenCommand
+ *
+ * @covers ::<!public>
+ * @covers ::__construct
+ */
+class CreateApiTokenCommandCaseTest extends AbstractCommandTestCase
```
Comment (Potherca) [inline src/Tests/Command/CreateApiTokenCommandTest.php]: Class name does not match filename.

File: src/Tests/Command/AbstractCommandTest.php [hunk 5]
```diff
@@ -0,0 +1,51 @@
...
+use Symfony\Component\Console\Input\StringInput;
+use Symfony\Component\Console\Output\StreamOutput;
+
+/**
+ * @coversDefaultClass \DealerDirect\Security\API\Command\AbstractCommand
+ *
+ * @covers ::<!public>
+ * @covers ::__construct
+ */
+class AbstractCommandTest extends AbstractCommandTestCase
```
Comment (Potherca) [inline src/Tests/Command/AbstractCommandTest.php]: Abstract class should be made abstract.

File: src/Tests/Command/CreateApiTokenCommandTest.php [hunk 6]
```diff
@@ -0,0 +1,124 @@
...
+{
+    /** @var $command AbstractCommand */
+    private $command;
+
+    public function setUp()
+    {
+        parent::setUp();
+
+        $this->command = new CreateApiTokenCommand();
```
Comment (Potherca) [inline src/Tests/Command/CreateApiTokenCommandTest.php]: Parent already has a `$command`  why not just use a setter?

File: src/Tests/Command/CreateApiTokenCommandTest.php [hunk 7]
```diff
@@ -0,0 +1,124 @@
...
+    }
+
+    /**
+     *
+     */
+    public function testCreateApiTokenCommandWithoutAnyInput()
```
Comment (Potherca) [inline src/Tests/Command/CreateApiTokenCommandTest.php]: Function name does not contain expectation.

File: src/Tests/Command/CreateApiTokenCommandTest.php [hunk 8]
```diff
@@ -0,0 +1,124 @@
...
+        $this->setupGeneratorCreateAPITokenMethod($mockAPIToken);
+
+        $mockAPIToken->method('getApiToken')->willReturn($mockToken);
+
+        $this->command->setApplication($mockApplication);
+        $this->command->run($input, $output);
+
+        $output = $this->retrieveOutput($tempFile);
+
+        $outputFormatter = new OutputFormatter();
```
Comment (Potherca) [inline src/Tests/Command/CreateApiTokenCommandTest.php]: Why u no use mock?

File: src/Tests/Command/ListApiTokensCommandTest.php [hunk 9]
```diff
@@ -0,0 +1,64 @@
...
+    {
+        parent::setUp();
+
+        $this->command = new ListApiTokensCommand();
+    }
+
+    /**
+     *
+     */
+    public function testListAllAPITokensWithoutTokens()
```
Comment (Potherca) [inline src/Tests/Command/ListApiTokensCommandTest.php]: Function name too generic. `testFooShouldBarWhenBaz`?

File: src/Tests/Command/AbstractCommandTest.php [hunk 10]
```diff
@@ -0,0 +1,51 @@
...
+{
+    /** @var $command AbstractCommand */
+    private $command;
+
+    public function setUp()
```
Comment (Potherca) [inline src/Tests/Command/AbstractCommandTest.php]: I would suggest making this function abstract as to force extending classes to implement it setting the `$command`.

**Comment by Potherca** [comment]
@brammittendorff-dd You seem to have missed quite some issues in your review...

File: phpunit.xml.dist [hunk 11]
```diff
@@ -0,0 +1,65 @@
...
+         verbose="true"
+         xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance"
+         xsi:noNamespaceSchemaLocation="http://schema.phpunit.de/5.7/phpunit.xsd"
+>
+    <php>
+        <ini name="error_reporting" value="-1"/>
+    </php>
+
+    <testsuites>
+        <testsuite name="Library">
```
Comment (hipio) [inline phpunit.xml.dist]: Fixed in upcoming PR.

File: src/Tests/Command/AbstractCommandTest.php (hunk 5 above)
Comment (hipio) [inline src/Tests/Command/AbstractCommandTest.php]: This tests the `AbstractCommand`-class. Will try and rename this test to make this more clear.

File: src/Tests/Command/AbstractCommandTest.php (hunk 10 above)
Comment (hipio) [inline src/Tests/Command/AbstractCommandTest.php]: Resolved with your help, thank you.

File: src/Tests/AbstractCommandTestCase.php (hunk 2 above)
Comment (hipio) [inline src/Tests/AbstractCommandTestCase.php]: I have everything correctly sorted in the upcoming PR. Feel free to check tomorrow.

File: src/Tests/Command/CreateApiTokenCommandTest.php (hunk 8 above)
Comment (hipio) [inline src/Tests/Command/CreateApiTokenCommandTest.php]: Because output is processed before I can read it so I must format my expectations as well. 😢

**Comment by hipio** [comment]
@Potherca All reviewed items without a comment have been changed as well. The next PR will include all these changes.

File: src/Tests/Command/CreateApiTokenCommandTest.php (hunk 8 above)
Comment (Potherca) [inline src/Tests/Command/CreateApiTokenCommandTest.php]: Will discuss in person.

**Comment by hipio** [comment]
All the changes have been made and missing unit tests have been added in #11

**Pull request merged**